src/panels/config/lovelace/dashboards/ha-config-lovelace-dashboards.ts

```diff
@@ -0,0 +1,270 @@
+                  "ui.panel.config.lovelace.dashboards.picker.open"
+                )}</mwc-button
+              >
+            `,
+        },
+      };
+      return { ...columns, ...columns2 };
+    }
+  );
+
+  private _getItems = memoize((dashboards: LovelaceDashboard[]) => {
+    return dashboards.map((dashboard) => {
+      return {
+        ...dashboard,
+        icon: dashboard.sidebar?.icon,
+        title: dashboard.sidebar?.title || dashboard.url_path,
+      };
+    });
+  });
+
+  protected render(): TemplateResult {
+    if (!this.hass || this._dashboards === undefined) {
+      return html`
+        <hass-loading-screen></hass-loading-screen>
+      `;
+    }
+
+    return html`
+      <hass-tabs-subpage-data-table
+        .hass=${this.hass}
+        .narrow=${this.narrow}
+        back-path="/config"
+        .route=${this.route}
+        .tabs=${lovelaceTabs}
+        .showAdvanced=${this.showAdvanced}
+        .columns=${this._columns(this.hass.language, this._dashboards)}
+        .data=${this._getItems(this._dashboards)}
+        @row-click=${this._openDialog}
+      >
+      </hass-tabs-subpage-data-table>
+      <ha-fab
+        ?is-wide=${this.isWide}
+        ?narrow=${this.narrow}
+        icon="hass:plus"
+        title="${this.hass.localize(
+          "ui.panel.config.lovelace.dashboards.picker.add_dashboard"
+        )}"
+        @click=${this._openDialog}
+      ></ha-fab>
+    `;
+  }
+
+  protected firstUpdated(changedProps: PropertyValues) {
+    super.firstUpdated(changedProps);
+    this._getDashboards();
+  }
+
+  private async _getDashboards() {
+    this._dashboards = await fetchDashboards(this.hass);
+  }
+
+  private _navigate(ev: Event) {
+    ev.stopPropagation();
+    const url = `/${(ev.target as any).urlPath}`;
+    navigate(this, url);
+  }
+
+  private async _openDialog(ev: CustomEvent): Promise<void> {
+    const id = (ev.detail as RowClickedEvent).id;
+    const dashboard = this._dashboards.find((res) => res.id === id);
+    if (!dashboard || dashboard.mode === "yaml") {
+      showAlertDialog(this, {
+        text: this.hass!.localize(
+          "ui.panel.config.lovelace.dashboards.cant_edit_yaml"
+        ),
+      });
+      return;
+    }
+    showDashboardDetailDialog(this, {
+      dashboard,
+      createDashboard: async (values: LovelaceDashboardCreateParams) => {
+        const created = await createDashboard(this.hass!, values);
+        this._dashboards = this._dashboards!.concat(
+          created
+        ).sort((res1, res2) => compare(res1.url_path, res2.url_path));
+      },
+      updateDashboard: async (values) => {
+        const updated = await updateDashboard(
+          this.hass!,
+          dashboard!.id,
+          values
+        );
+        this._dashboards = this._dashboards!.map((res) =>
+          res === dashboard ? updated : res
+        );
+      },
+      removeDashboard: async () => {
+        if (
+          !(await showConfirmationDialog(this, {
```

Comment:
I would expect confirmation to be done in the dialog

Reply:
The benefit of this is that we could add delete buttons in the data-table in the future and use the same logic. Not sure if we want to do that.